subprojects/core-api/src/main/java/org/gradle/api/file/LinksStrategy.java

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright 2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.api.file;
+
+import org.gradle.api.GradleException;
+import org.gradle.api.Incubating;
+
+import javax.annotation.Nullable;
+import java.io.IOException;
+import java.io.Serializable;
+import java.nio.file.Files;
+import java.nio.file.Path;
+
+/**
+ * Strategy for handling symbolic links when copying files.
+ *
+ * @since 8.6
+ */
+@Incubating
+public interface LinksStrategy extends Serializable {
+    /**
+     * Do not preserve any symlinks. This is the default.
```

Comment:
Could you please clarify on what exactly "not preserve" means? Does it mean links are completely ignored, or that they are replaced with copies?

Depending on that answer, would it make sense to add another strategy that *does* replace links with copies, e.g. to support extracting ZIP archives to file systems that do not support symbolic links?

Finally, if this is the default in order to not introduce a breaking behavior because it resembles what has been the only strategy so far (before Gradle 8.6), that should be documented as well IMO.

Reply:
Thanks for the feedback!

"Do not preserve" means treat any link as a file it points to. In other words, replacing with copies. This is the default behavior, and this is how Gradle works in earlier versions. 

This is [documented](https://github.com/gradle/gradle/pull/25264/files#diff-125f4d7aea81fa9326b07bbe9669d1f1c3ea15dc1e4984f783a943e5e0f0b9a8R675) in the regular docs, but I'll update this Javadoc comment accordingly.